packages/legend-extension-external-language-morphir/src/components/ELMorphir_StudioPlugin.tsx

```diff
@@ -0,0 +1,200 @@
+ */
+
+import packageJson from '../../package.json';
+
+import type {
+  DSL_StudioPlugin_Extension,
+  GenerationFile,
+  FileGenerationState,
+  StudioPluginManager,
+  FileGenerationResultViewerAction,
+  FileGenerationScopeFilter,
+} from '@finos/legend-studio';
+
+import { StudioPlugin } from '@finos/legend-studio';
+import {
+  NetworkClient,
+  assertNonEmptyString,
+  guaranteeNonEmptyString,
+} from '@finos/legend-shared';
+import type { PackageableElement } from '@finos/legend-graph';
+import { ConcreteFunctionDefinition } from '@finos/legend-graph';
+
+const MORPHIR_TYPE_NAME = `morphir`;
+
+interface ELMorphir_GenerationPluginConfigData {
+  morphirVisualizerUrl: string;
+  linterServerUrl: string;
+  linterAppUrl: string;
+}
+
+export class ELMorphir_GenerationPlugin
+  extends StudioPlugin
+  implements DSL_StudioPlugin_Extension
+{
+  networkClient: NetworkClient;
+  private _morphirVisualizerUrl?: string | undefined;
+  private _linterServerUrl?: string | undefined;
+  private _linterAppUrl?: string | undefined;
+
+  constructor() {
+    super(packageJson.extensions.studioPlugin, packageJson.version);
+    this.networkClient = new NetworkClient();
+  }
+
+  install(pluginManager: StudioPluginManager): void {
+    pluginManager.registerStudioPlugin(this);
+  }
+
+  override configure(_configData: object): ELMorphir_GenerationPlugin {
+    const configData = _configData as ELMorphir_GenerationPluginConfigData;
+    assertNonEmptyString(
+      configData.morphirVisualizerUrl,
+      `Can't configure morphir visualizer url for generation plugin: 'url' field is missing or empty`,
+    );
+    assertNonEmptyString(
+      configData.linterServerUrl,
+      `Can't configure linter server url for generation plugin: 'url' field is missing or empty`,
+    );
+    assertNonEmptyString(
+      configData.linterAppUrl,
+      `Can't configure linter app url for generation plugin: 'url' field is missing or empty`,
+    );
+    this.morphirVisualizerUrl = configData.morphirVisualizerUrl;
+    this.linterServerUrl = configData.linterServerUrl;
+    this.linterAppUrl = configData.linterAppUrl;
+
+    return this;
+  }
+
+  set morphirVisualizerUrl(url: string) {
+    this._morphirVisualizerUrl = url;
+  }
+
+  get morphirVisualizerUrl(): string {
+    return guaranteeNonEmptyString(this._morphirVisualizerUrl);
+  }
+
+  set linterServerUrl(url: string) {
+    this._linterServerUrl = url;
+  }
+
+  get linterServerUrl(): string {
+    return guaranteeNonEmptyString(this._linterServerUrl);
+  }
+
+  set linterAppUrl(url: string) {
+    this._linterAppUrl = url;
+  }
+
+  get linterAppUrl(): string {
+    return guaranteeNonEmptyString(this._linterAppUrl);
+  }
+
+  getExtraFileGenerationResultViewerActions(): FileGenerationResultViewerAction[] {
+    const visualizeMorphir =
+      (fileNode: GenerationFile): (() => void) =>
+      async (): Promise<void> => {
+        window.open(this.morphirVisualizerUrl);
+        await this.networkClient.post(
```

Comment:
you make the POST call after opening the app?

Reply:
Moved opening window to post POST call.